extension/goplugin/schemas_test.go

```diff
@@ -304,4 +304,62 @@ var _ = Describe("Schemas", func() {
 		})
 	})
 
+	Context("Convert", func() {
+		It("should convert context to resource", func() {
+			context := map[string]interface{}{
+				"id":          "42",
+				"description": "test",
+				"subobject": map[string]interface{}{
+					"subproperty": "testproperty",
+				},
+			}
+			expected := &test.Test{
+				ID:          "42",
```

Comment:
please add more test cases:
1. nil/null instead of subobject (both ways: convertTo/convertFrom)
2. negative cases (like int instead of string, null where it's not allowed)